src/main/kotlin/com/expedia/graphql/schema/hooks/DataFetcherExecutionPredicate.kt

```diff
@@ -0,0 +1,18 @@
+package com.expedia.graphql.schema.hooks
+
+interface DataFetcherExecutionPredicate {
+
+    /**
+     * The test to perform on the argument.
+     *
+     * With this test, runtime evaluation can be performed such as validation
+     */
+    fun <T> test(value: T): Boolean
```

Comment:
Should we follow the `will` `did` pattern instead here and have two fields on the hooks?